gensim/models/word2vec_corpusfile.pyx

```diff
@@ -186,7 +186,8 @@ cdef void prepare_c_structures_for_batch(
         vector[vector[string]] &sentences, int sample, int hs, int window, long long *total_words,
         int *effective_words, int *effective_sentences, unsigned long long *next_random,
         cvocab_t *vocab, int *sentence_idx, np.uint32_t *indexes, int *codelens,
-        np.uint8_t **codes, np.uint32_t **points, np.uint32_t *reduced_windows) nogil:
+        np.uint8_t **codes, np.uint32_t **points, np.uint32_t *reduced_windows,
+        bint do_reduced_windows) nogil:
```

Comment:
GIven that there's an existing variable, `reduced_windows`, which is an array of the precomputed sampled effective windows (something simply called `b` in the original `word2vec.c` source code IIRC), we definitely should *not* be naming the top-level toggle parameter the same thing almost everywhere else, but then change that toggle to `do_reduced_windows` here. That's a recipe for unnecessary confusion. The top-level param should be a name that is (1) different; (2) consistent. Perhaps a change as small as `reduce_windows` (imperative sense) is enough; but for more contrast & descriptiveness, maybe the parameter should be named `shrink_windows`.

Reply:
Also: our other de facto boolean toggles - like `hs` and `sample` here – have otherwise been passed as type `int`, not `bint`. Even if `bint` might be better, I'd prefer consistency.

Reply:
We (M-Demay and I) used this name based on the discussion in the associated issue, but we both agree that is was not such a great choice. The next commit goes for `shrink_windows`.
(and ok for using `int` rather than `bint`)